csrc/root_domain_map.cpp

```diff
@@ -447,12 +447,44 @@ bool UnmappableReductionDomains::isReductionOutputMapped(
               input_keys.end();
         });
     // None of the consumer domains is used for the reduction
-    // domain. They should be safe with respect to this reduction
-    // domain
+    // domain.
+    if (it == consumer_domains.end()) {
+      // Further check if any consumer domain is mapped with the reduction input
+      // domains
+      it = std::find_if(
```

Comment:
Let's merge this `std::find_if` with the above `std::find`

Reply:
merged.